merkledag/ipld.md

```diff
@@ -72,6 +83,81 @@ O_5 = | "hello": "world"  |  whose hash value is QmR8Bzg59Y4FGWHeu9iTYhwhiP8PHCN
 
 This entire _merkle-path_ traversal is a unix-style path traversal over a _merkle-dag_ which uses _merkle-links_ with names.
 
+### IPLD merkle-path (best solution)
+
+An _IPLD merkle-path_ is an extension of a _filesystem merkle-path_ which uses a special syntax to access link properties.
+
+Path elements are suffixed by either `.link` to access the link properties or by `.object` to dereference the _merkle-link_. if no suffix is present, the _merkle-link_ is dereferenced (to be compatible with _filesystem merkle-paths_ in most cases)
+
+**FIXME**: perhaps use different suffixes so we are less likely to have ambiguities. Using a character that is denied by Windows would be a good idea since those are less likely to be present in filenames. For most cases, this would make _IPLD merkle-paths_ a superset of _filesystem merkle-paths_. For example we could use `?link` and `?object`
+
+Suppose we have object which hashes to QmCCC...000:
+
+    ---
+    stuff:
+      foo:
+        mlink: QmCCC...111
+        mode: 0755
+        owner: jbenet
+
+and we have object which hashes to QmCCC...111 (the foo link):
+
+    ---
+    other:
+      cat.link:
+        mlink: QmCCC...222
+        mode: 0644
+        owner: jbenet
+
+Now:
+
+- the path `/ipfs/QmCCC...000/stuff/foo.link/mode` yields `0755`
+- the path `/ipfs/QmCCC...000/stuff/foo/other/cat.link/mode` does not exists because `other` does not have a `cat` object, only a `cat.link`
+- the path `/ipfs/QmCCC...000/stuff/foo/other/cat.link.link/mode` yields `0644`
+- the path `/ipfs/QmCCC...000/stuff/foo.object/other/cat.link` yields object `QmCCC...222`
+
+### IPLD merkle-path (other solution)
+
+An _IPLD merkle-path_ is a path which initially dereferences through a _merkle-link_ and then follows elements in intermediate objects through the separator `.`, and follows _merkle-links_ through the separator `/`.
+
+**Variation:** The separator `/` can also be used instead of `.` if there is no ambiguity.
+
+The separator can be escaped in any path element using `\.`, and the `\` character is escaped using `\\`.
+
+Suppose we have object which hashes to QmCCC...000:
+
+    ---
+    stuff:
+      foo:
+        mlink: QmCCC...111
+        mode: 0755
+        owner: jbenet
+
+and we have object which hashes to QmCCC...111 (the foo link):
+
+    ---
+    other:
+      cat.jpg:
+        mlink: QmCCC...222
+        mode: 0644
+        owner: jbenet
+
+Now:
+
+- the path `/ipfs/QmCCC...000/stuff.foo.mode` yields `0755`
+- the path `/ipfs/QmCCC...000/stuff/foo` does not exists because in the object `QmCCC...000`, the `stuff` object cannot is not a _merkle-link_ (it doesn't have the `mlink` key)
+- the path `/ipfs/QmCCC...000/stuff.foo/other.cat\.jpg` yields object `QmCCC...222`
```

Comment:
- using `.` requiring escape is liable to confuse a lot of people since dots are used for extensions :/

(i do agree that we should avoid ambiguous meanings, like `orther.car.jpg` is totally weird)